extensions/WindowsSandboxExtension/Providers/WindowsSandboxComputeSystem.cs

```diff
@@ -0,0 +1,204 @@
+    private ComputeSystemState _state = ComputeSystemState.Stopped;
+
+    private ComputeSystemState State
+    {
+        get => _state;
+
+        set
+        {
+            _state = value;
+            StateChanged?.Invoke(this, value);
+        }
+    }
+
+    public string AssociatedProviderId => Constants.ProviderId;
+
+    public string DisplayName => Resources.GetResource("WindowsSandboxDisplayName", _log);
+
+    public string Id => _id.ToString();
+
+    public string SupplementalDisplayName => string.Empty;
+
+    public ComputeSystemOperations SupportedOperations => ComputeSystemOperations.Terminate;
+
+    public IDeveloperId? AssociatedDeveloperId => null;
+
+    public event TypedEventHandler<IComputeSystem, ComputeSystemState>? StateChanged;
+
+    public IAsyncOperation<ComputeSystemThumbnailResult> GetComputeSystemThumbnailAsync(string options)
+    {
+        return Task.Run(async () =>
+        {
+            var uri = new Uri(Constants.Thumbnail);
+            var storageFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+            var randomAccessStream = await storageFile.OpenReadAsync();
+
+            // Convert the stream to a byte array
+            var bytes = new byte[randomAccessStream.Size];
+            await randomAccessStream.ReadAsync(bytes.AsBuffer(), (uint)randomAccessStream.Size, InputStreamOptions.None);
+            return new ComputeSystemThumbnailResult(bytes);
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<IEnumerable<ComputeSystemProperty>> GetComputeSystemPropertiesAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            var properties = new List<ComputeSystemProperty>
+            {
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.CpuCount, Environment.ProcessorCount),
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.AssignedMemorySizeInBytes, 4294967296),
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.StorageSizeInBytes, 85899345920),
+                ComputeSystemProperty.Create(ComputeSystemPropertyKind.UptimeIn100ns, 100),
+            };
+
+            return properties.AsEnumerable();
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemStateResult> GetStateAsync()
+    {
+        return Task.Run(() =>
+        {
+            return new ComputeSystemStateResult(State);
+        }).AsAsyncOperation();
+    }
+
+    public IAsyncOperation<ComputeSystemOperationResult> ConnectAsync(string options)
+    {
+        return Task.Run(() =>
+        {
+            try
+            {
+                lock (_windowsSandboxStartLock)
+                {
+                    // Windows Sandbox is not running.
+                    if (_windowsSandboxExeProcess == null || _windowsSandboxExeProcess.HasExited)
+                    {
+                        var system32Path = Environment.GetFolderPath(Environment.SpecialFolder.System);
```

Comment:
can we do `State = ComputeSystemState.Starting;` here first? this way the UI will say its starting beforehard.